sdks/python/apache_beam/transforms/util_test.py

```diff
@@ -908,7 +908,7 @@ def test_in_global_window_with_text_file(self):
     with tempfile.NamedTemporaryFile(suffix=".json") as f:
       with open(f.name, "w") as fh:
         json.dump(GroupIntoBatchesTest._create_test_data(), fh)
-      with self.assertRaises(RuntimeError):
+      with self.assertRaises((RuntimeError, AttributeError)):
```

Comment:
I see, this means GroupIntoBatches still cannot working in direct runner. It now throws AttributeError.

assertRaises here actually tests the feature gap for sdf_direct_runner exists

In this sense think unit test belongs to sdf_direct_runner_test:

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/direct/sdf_direct_runner_test.py

Reply:
Fine. The test still fails on GitHub Action windows:

```
    def test_in_global_window_with_text_file(self):
      with tempfile.NamedTemporaryFile(suffix=".json") as f:
>       with open(f.name, "w") as fh:
E       PermissionError: [Errno 13] Permission denied: 'D:\\a\\beam\\beam\\sdks\\python\\target\\.tox\\py37-win\\tmp\\tmphcz3z87a.json'
```

a simpler test is suggested here: https://github.com/apache/beam/issues/26836#issuecomment-1573128291

Reply:
We should then leave some comment to the added test sth like

# TODO sdf_direct_runner currently does not support GroupIntoBatches and will throw this AttributeError

Reply:
I mean we no longer need `test_in_global_window_with_text_file`. It fails on Github Action

Reply:
Let me fix that test for windows. More tests are better. In particular, it tests reading the actual file.

Reply:
this is unit test. Not neessary. Real file system is unpredictable (like here)

Reply:
I know. But the tests are also great sources for examples.